src/compiler/checker.ts

```diff
@@ -10403,6 +10433,7 @@ namespace ts {
                 case SyntaxKind.TemplateExpression:
                     return checkTemplateExpression(<TemplateExpression>node);
                 case SyntaxKind.StringLiteral:
+                    return checkStringLiteralExpression(<LiteralExpression>node);
                 case SyntaxKind.NoSubstitutionTemplateLiteral:

```

Comment:
To maybe clarify:
```
type CardinalDirection = "North" | "East" | "South" | "West";
```
Gets encoded as:
```
let north = [2, 32];
let east = [2, 64];
let south = [2, 128];
let west = [2, 256];

let CardinalDirection = [[2, 480]];
```
Very easy to check if 'north' belongs in ```CardinalDirection```